src/main/java/org/webshar/hrms/repository/LeaveAllocationRepository.java

```diff
@@ -33,14 +28,25 @@
       @Param("start_date") final Date startDate,
       @Param("end_date") final Date endDate
   );
-  public LeaveAllocation findByEmployeeIdAndLeaveTypeId(
-      @Param("employee_id") final Long employeeId,
-      @Param("leave_type_id") final Long leaveTypeId
-  );
 
   public void deleteByEmployeeId(@Param("employee_id") final Long employeeId);
 
   public void deleteByEmployeeIdAndLeaveTypeId(@Param("employee_id") final Long employeeId,
       @Param("leave_type_id") final Long leaveTypeId);
 
+  @Query
+      (" SELECT l " +
+      " FROM LeaveAllocation l " +
+      " WHERE l.employee.id = :employeeId" +
+      " AND l.leaveType.id = :leaveTypeId" +
+      " AND (" +
+      "       (:startDate <= l.startDate AND l.startDate <=  :endDate)" +
+      "       OR (:startDate <= l.endDate AND l.endDate <=  :endDate)" +
+      "       OR (l.startDate <= :startDate AND  :endDate <= l.endDate )" +
+      " )")
+  public LeaveAllocation findGivenTypeOfLeaveAllocatedOrNotInTheGivenDateRange(
+      @Param("employeeId") Long employeeId,
+      @Param("leaveTypeId") Long leaveTypeId,
+      @Param("startDate") LocalDate startDate,
+      @Param("endDate") LocalDate endDate);
```

Comment:
The return type of this response should `List<LeaveAllocation>`. As there could be possibility of finding multiple rows.